packages/devtools_app/test/memory/allocation_profile/allocation_profile_table_view_test.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:devtools_app/src/config_specific/import_export/import_export.dart';
+import 'package:devtools_app/src/screens/memory/memory_controller.dart';
+import 'package:devtools_app/src/screens/memory/memory_heap_tree_view.dart';
+import 'package:devtools_app/src/screens/memory/memory_screen.dart';
+import 'package:devtools_app/src/screens/memory/panes/allocation_profile/allocation_profile_table_view_controller.dart';
+import 'package:devtools_app/src/service/service_manager.dart';
+import 'package:devtools_app/src/shared/globals.dart';
+import 'package:devtools_app/src/shared/preferences.dart';
+import 'package:devtools_shared/devtools_shared.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../test_data/memory_allocation.dart';
+
+void main() {
+  late MemoryController controller;
+  late FakeServiceManager fakeServiceManager;
+
+  void _setUpServiceManagerForMemory() {
+    // Load canned data testHeapSampleData.
+    final allocationJson =
+        AllocationMemoryJson.decode(argJsonString: testAllocationData);
+
+    fakeServiceManager = FakeServiceManager(
+      service: FakeServiceManager.createFakeService(
+        allocationData: allocationJson,
+      ),
+    );
+    when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isFlutterAppNow).thenReturn(true);
+    when(fakeServiceManager.connectedApp!.isDartCliAppNow).thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isDebugFlutterAppNow)
+        .thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isDartWebApp)
+        .thenAnswer((_) => Future.value(false));
+    setGlobal(ServiceConnectionManager, fakeServiceManager);
+  }
+
+  Future<void> pumpMemoryScreen(
+    WidgetTester tester, {
+    MemoryController? memoryController,
+  }) async {
+    await tester.pumpWidget(
+      wrapWithControllers(
+        const MemoryBody(),
+        memory: controller = memoryController ?? MemoryController(),
+      ),
+    );
+
+    // Delay to ensure the memory profiler has collected data.
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(find.byType(MemoryBody), findsOneWidget);
+  }
+
+  // Set a wide enough screen width that we do not run into overflow.
+  const windowSize = Size(2225.0, 1000.0);
+
+  test('Allocation profile disabled by default', () {
+    // TODO(bkonyi): remove this check once we enable the tab by default.
+    expect(enableNewAllocationProfileTable, isFalse);
+  });
+
+  group('Allocation Profile Table', () {
+    setUpAll(() => enableNewAllocationProfileTable = true);
+    tearDownAll(() => enableNewAllocationProfileTable = false);
+
+    setUp(() async {
+      fakeServiceManager = FakeServiceManager();
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
```

Comment:
we are doing this twice - once here and once in `_setUpServiceManagerForMemory`. This may be confusing to keep track of which FakeServiceManager is in use, so let's consolidate if possible.